documents/xaml-suggested-actions/xaml-designer-suggested-actions-extensibility.md

```diff
@@ -0,0 +1,115 @@
+## Suggested Actions Extensibility
+
+- [Tokens](./xaml-designer-suggested-actions-extensibility-tokens.md)
+- [Features](./xaml-designer-suggested-actions-extensibility-features.md)
+- [Behaviors](./xaml-designer-suggested-actions-extensibility-behaviors.md)
+- [Customization](./xaml-designer-suggested-actions-extensibility-customization.md)
+
+In Visual Studio 2019 16.7 Preview 3 we added extensibility support for "XAML Suggested Actions"
+>*Note: this API will be available only in Preview Channel until feature is finalized and completed*
+
+To enable "Xaml Suggested Actions" for any control, a `SuggestedActionProvider` feature provider should be created and registered in metadata.
+
+>*A Control could have multiple Suggested Actions providers. Each provider will be shown as a separate tab in the Suggested Actions UI.*
+
+Example:
+```cs
+public class ExampleButton : Button { }
+```
+
+![extensibility-migration-architecture](xaml-suggested-actions.png)
+
+### Implementation for example above:
+```CS
+public class ExampleButtonSuggestedActionProvider : SuggestedActionProvider
+{
+    public static ActionToken Token_Property_IsCancel = new ActionToken(0x1001);
+    public static ActionToken Token_Property_IsDefault = new ActionToken(0x1002);
+    public static ActionToken Token_Last = new ActionToken(0x10FF);
+    public override string Header => "Actions";
+    public override string Type => "CustomControlLibrary.WpfCore.ExampleButton";
+    public override void Initialize()
+    {
+        this.ShowNameProperty = true;
+        base.Initialize();
+        
+        this.AddGroup(new ActionGroup(SuggestedActionProviderTokens.Token_Group_Common,
+            new PropertyAction(SuggestedActionProviderTokens.Token_Property_Content, "Content"),
+            new PropertyAction(SuggestedActionProviderTokens.Token_Property_Background, "Background"),
+            new PropertyAction(SuggestedActionProviderTokens.Token_Property_IsEnabled, "IsEnabled")
+            ));
+        
+        this.AddGroup(new ActionGroup(SuggestedActionProviderTokens.Token_Group_Specific,
+            new PropertyAction(ExampleButtonSuggestedActionProvider.Token_Property_IsCancel,"IsCancel"),
+            new PropertyAction(ExampleButtonSuggestedActionProvider.Token_Property_IsDefault,"IsDefault")));
+        
+        this.AddGroup(new BorderBrushActionGroup());
+        this.AddGroup(new VisibilityActionGroup());
+        this.AddGroup(new FontSettingsActionGroup());
+    }
+}
+```
+### Example for inherited controls
```

Comment:
Perhaps you could add a sentence here explaining how the provider can modify the base behavior. I had to look back and forth a few time to notice the differences in the image.